src/finufft.cpp

```diff
@@ -568,6 +594,12 @@ int FINUFFT_MAKEPLAN(int type, int dim, BIGINT *n_modes, int iflag, int ntrans,
     fprintf(stderr, "[%s] ntrans (%d) should be at least 1.\n", __func__, ntrans);
     return FINUFFT_ERR_NTRANS_NOTVALID;
   }
+  if (!p->opts.fftw_lock_fun != !p->opts.fftw_unlock_fun) {
```

Comment:
Should't we provide a basic locking mechanism even if the user does not pass a lock?